pkg/networkservice/chains/nsmgr/suite_test.go

```diff
@@ -394,49 +398,57 @@ func (s *nsmgrSuite) Test_PassThroughRemoteUsecase() {
 	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
 	defer cancel()
 
+	nsReg := linearNS(nodesCount)
+	nsReg, err := s.domain.Nodes[0].NSRegistryClient.Register(ctx, nsReg)
+	require.NoError(t, err)
+
 	var nsesReg [nodesCount]*registry.NetworkServiceEndpoint
 	for i := 0; i < nodesCount; i++ {
-		var additionalFunctionality []networkservice.NetworkServiceServer
-		if i != 0 {
-			// Passtrough to the node i-1
-			additionalFunctionality = []networkservice.NetworkServiceServer{
-				chain.NewNetworkServiceServer(
-					clienturl.NewServer(s.domain.Nodes[i].NSMgr.URL),
-					connect.NewServer(ctx,
-						sandbox.NewCrossConnectClientFactory(sandbox.GenerateTestToken,
-							newPassTroughClient(fmt.Sprintf("my-service-remote-%v", i-1)),
-							kernel.NewClient()),
-						connect.WithDialTimeout(sandbox.DialTimeout),
-						connect.WithDialOptions(sandbox.DefaultDialOptions(sandbox.GenerateTestToken)...),
-					),
-				),
-			}
-		}
-
-		nsReg, err := s.domain.Nodes[0].NSRegistryClient.Register(ctx, &registry.NetworkService{
-			Name: fmt.Sprintf("my-service-remote-%v", i),
-		})
-		require.NoError(t, err)
-
 		nsesReg[i] = &registry.NetworkServiceEndpoint{
 			Name:                fmt.Sprintf("endpoint-%v", i),
 			NetworkServiceNames: []string{nsReg.Name},
+			NetworkServiceLabels: map[string]*registry.NetworkServiceLabels{
+				nsReg.Name: {
+					Labels: map[string]string{
+						step: fmt.Sprintf("%v", i),
+					},
+				},
+			},
+		}
+
+		var additionalFunctionality []networkservice.NetworkServiceServer
+		if i != nodesCount-1 {
+			additionalFunctionality = additionalFunctionalityChain(ctx, s.domain.Nodes[i].NSMgr.URL,
+				fmt.Sprintf("endpoint-client-%v", i), nsesReg[i].NetworkServiceLabels[nsReg.Name].Labels)
 		}
+
 		_, err = s.domain.Nodes[i].NewEndpoint(ctx, nsesReg[i], sandbox.GenerateTestToken, additionalFunctionality...)
 		require.NoError(t, err)
 	}
 
-	nsc := s.domain.Nodes[nodesCount-1].NewClient(ctx, sandbox.GenerateTestToken)
+	nsc := s.domain.Nodes[0].NewClient(ctx, sandbox.GenerateTestToken)
 
-	request := defaultRequest(fmt.Sprintf("my-service-remote-%v", nodesCount-1))
+	request := defaultRequest(nsReg.Name)
 
+	// Request
 	conn, err := nsc.Request(ctx, request)
 	require.NoError(t, err)
 	require.NotNil(t, conn)
 
 	// Path length to first endpoint is 5
 	// Path length from NSE client to other remote endpoint is 8
 	require.Equal(t, 8*(nodesCount-1)+5, len(conn.Path.PathSegments))
+	for i := 0; i < len(nsReg.Matches); i++ {
+		if i == 0 {
```

Comment:
I guess that there is no need for the additional _if 0_ case :)
(0 + 1) * 5 - 1 = 4
0 * 8 + 4 = 4